app/controllers/concerns/params_store_concern.rb

```diff
@@ -0,0 +1,15 @@
+module ParamsStoreConcern
+  extend ActiveSupport::Concern
+
+  def store_params
+    return if session[:stored_params].present?
```

Comment:
t'es sur qu'on ne veut pas écraser avec la dernière simulation par exemple ?

Reply:
Sans la guard clause, le problème qui se pose est le suivant : 

1. Dans une fenêtre de navigation privée, je vais sur `/commencer/test-2?champ_Q2hhbXAtMg=texte` => ça stocke les paramètres en passant dans `store_params`
2. Je m'authentifie (sign up ou sign in, peu importe) => ça me renvoie vers `/commencer/test-2`, **sans les paramètres**, et donc `store_params` va effacer les paramètres qui étaient précédemment stockés en session
3. Le dossier n'est pas prérempli :disappointed: 

Il y a peut-être moyen de conserver les paramètres lors de l'authentification (et donc de contourner ce problème), en modifiant un peu `store_user_location!`, mais sur le moment ça ne m'a pas semblé important. Est-ce que tu penses que ça vaut la peine que je creuse ?

Reply:
nop, peut etre un commentaire pour comprendre le raisonnement mais facultatif

Reply:
:heavy_check_mark: